### ISSUE TITLE
Reporting "won't fix" erroneously for certain Debian vulnerabilities

### ISSUE THREAD

**Issue by luhring** [anchore/grype#386]
**What happened**: Grype reports `wont-fix` in its JSON output's `.vulnerability.fix.state` value for Debian vulnerability records where the upstream Debian data doesn't appear to claim that the vulnerability won't be fixed. (For example, this is happening with the following Debian record: https://security-tracker.debian.org/tracker/CVE-2021-33574.)

**What you expected to happen**: Grype should only ever show `wont-fix` for vulnerabilities where the vendor has **explicitly** stated that the vulnerability will not be fixed.

**How to reproduce it (as minimally and precisely as possible)**:

1. Look at the vulnerability details for CVE-2021-33574, such as reported for the image `httpd:2.4`:

```
grype httpd:2.4 -o json | jq '.matches | map(.vulnerability | select(.id == "CVE-2021-33574"))[0]'
```

Output:

```json
{
  "id": "CVE-2021-33574",
  "dataSource": "https://security-tracker.debian.org/tracker/CVE-2021-33574",
  "namespace": "debian:10",
  "severity": "High",
  "urls": [
    "https://security-tracker.debian.org/tracker/CVE-2021-33574"
  ],
  "cvss": [],
  "fix": {
    "versions": [],
    "state": "wont-fix"
  },
  "advisories": []
}
```

Note the `"wont-fix"` value for the `.fix.state` field.

2. Look at the record in Debian directly: https://security-tracker.debian.org/tracker/CVE-2021-33574. This record **doesn't say** that this vulnerability won't be fixed.

**Anything else we need to know?**:

Here's a theory about what's causing this...

#### For context, here's the vulnerability "data pipeline" at play here:

`Debian data` -> `Anchore Feed Service` -> `Grype DB Builder` -> `Grype's database` -> `Grype`

#### How the Debian record for CVE-2021-33574 (as an example) is being transformed along this pipeline:

1. The `Debian data` record ([link](https://security-tracker.debian.org/tracker/CVE-2021-33574)) has **Notes** that include the "state" of `<no-dsa>`.

2. The `Anchore Feed Service` is representing this vulnerability record as this:

```jsonc
{
  "Vulnerability": {
   "CVSS": [],
   "Description": "",
   "FixedIn": [
    {
     "Name": "glibc",
     "NamespaceName": "debian:10",
     "VendorAdvisory": {
      "NoAdvisory": true
     },
// ...
```

(Notice the `"NoAdvisory": true`.)

This appears to be the result of [this logic](https://github.com/anchore/enterprise/blob/master/src/anchore_enterprise/services/feeds/drivers/debian/data.py#L473-L479) (internal link) which reacts to the presence of `<no-dsa>` in the Debian notes.

3. Per the guidance in item 1 of [this issue note](https://github.com/anchore/grype/issues/276#issue-849520824), Grype's DB building process is translating `"NoAdvisory": true` into the `wont-fix` value.

#### The problem

Looking at [this documentation from the Debian Security Team](https://security-team.debian.org/security_tracker.html#issues-not-warranting-a-security-advisory), we believe `<no-dsa>` is not intended to be synonymous with "won't fix". Instead, `<ignored>` seems to mean "won't fix", whereas `<no-dsa>` might later be changed to `<ignored>`, or it may not.

If by `"NoAdvisory": true` the Feed Service is communicating a "won't fix" state, than this looks like a bug in the Feed Service.

If instead, `"NoAdvisory": true` is meant to indicate just that there isn't a Debian Security Advisory, then it's a bug that the Grype DB building process is translating this value to the `wont-fix` state.

**Environment**:
- Output of `grype version`:
- OS (e.g: `cat /etc/os-release` or similar):

**Comment by luhring** [comment]
For more context from a community member: https://anchorecommunity.slack.com/archives/C027JE5M345/p1629051686009200

**Comment by luhring** [comment]
@nightfurys Can you take a look at this and provide your assessment? Let me know if it's easier to handle via a Zoom sync 😃

**Comment by luhring** [comment]
**Thought:** It might be valuable for the data pipeline to allow the expression of _both_ dimensions simultaneously:

1. whether there's a DSA available, and
2. whether the issue won't be fixed
 
Currently, the neither the Feed Service nor Grype's DB infrastructure are capable of representing both of these two concepts.

**Comment by zhill** [comment]
I agree that indicating "will not issue advisory" and "will not fix" are different, but those can be subtle for each distro. We can see if there is a DSA/Advisory regardless of their state flag, and per their spect the fix may be in another DSA so it doesn't mean there will not be one, just that the CVE will not have its own DSA.

I'm in favor of fixing it to implement the will-no-fix semantics explicitly. An additional debian-specific field for no-dsa could be present in metadata since it isn't consistent for all data sources but would allow filtering using jq/templating potentially if the user knows the image is debian.

**Comment by luhring** [comment]
Is there a distro-agnostic solution here that accommodates the Debian `<no-dsa>` scenario? I'm thinking if the Feed Service's output was modified to have two fields, we'd be all set:

1. "is there an advisory present?" (I think this would correspond to the existing `NoAdvisory` boolean field)
2. "has the vendor said they **won't fix** this issue?" (new field, such as `VendorWontFix`)

The Feed Service's output could be adjusted to include item 2, and then it could implement the correct logic for both items 1 and 2.
